src/families.h

```diff
@@ -47,68 +45,65 @@ class Family {
     step_sizes.reserve(alpha_scaled.size());
 
     for (auto alpha_val : alpha_scaled) {
-      double L = L_scaling*(max_squared_sum + fit_intercept) + alpha_val;
+      double L =
+        L_scaling_*(max_squared_sum + static_cast<double>(fit_intercept))
+        + alpha_val;
       double mu_n = 2.0*n_samples*alpha_val;
       step_sizes.push_back(1.0 / (2.0*L + std::min(L, mu_n)));
     }
     return step_sizes;
-  };
-
-  std::size_t GetNClasses() {
-    return n_classes;
   }
 
+  unsigned n_classes() { return n_classes_; }
+  double lambda_scaling() { return lambda_scaling_; }
+
 protected:
-  std::size_t n_classes;
-  double L_scaling;
+  unsigned n_classes_;
+  double L_scaling_;
+  double lambda_scaling_;
 };
 
 class Gaussian : public Family {
 public:
   Gaussian() {
-    n_classes = 1;
-    L_scaling = 1.0;
-  };
+    n_classes_ = 1;
+    L_scaling_ = 1.0;
+    lambda_scaling_ = 1.0;
+  }
 
-  virtual ~Gaussian() {};
+  double Link(const double y) { return y; }
 
   double Loss(const double prediction, const double y) {
     return 0.5*(prediction - y)*(prediction - y);
-  };
+  }
 
   double Gradient(const double prediction, const double y) {
     return prediction - y;
-  };
+  }
 
   void PreprocessResponse(std::vector<double>& y,
                           std::vector<double>& y_center,
-                          std::vector<double>& y_scale,
-                          const bool           fit_intercept) {
-    if (fit_intercept) {
-      double y_mu = Mean(y, y.size());
-      double y_sd = StandardDeviation(y, y.size());
-
-      y_center.push_back(y_mu);
-      y_scale.push_back(y_sd);
-
-      for (auto& y_val : y) {
-        y_val -= y_mu;
-
-        if (y_sd != 0)
-          y_val /= y_sd;
-      }
-    } else {
-      y_center.push_back(0.0);
-      y_scale.push_back(1.0);
+                          std::vector<double>& y_scale) {
+    double y_mu = Mean(y);
+    double y_sd = StandardDeviation(y);
+
+    if (y_sd == 0.0) y_sd = 1.0;
```

Comment:
Does everything still work if `y` is constant? If it doesn't, then we just have `beta = 0` and `intercept = avg(y)` so we can special case this easily.

Reply:
We are using uncorrected standard deviation so are avoiding division by 0 in this case so this is not a problem (if I understood you correctly?).

Reply:
I was just wondering if everywhere else in the algorithm works fine with constant y. 

This would make a good (and easy) test case: if `y` is constant, then we should have `beta = 0`, `intercept = mean(y)` for the Gaussian. 

For the logistic, I think the intercept would diverge to `+/- infinity`, again with all `beta = 0`.

Reply:
Yeah, good thing you pointed it out. Computing the lambda path of course failed because lambda_max was initialized to 0. I have changed it so that a path of zeroes is returned in this case instead.

```r
x <- as.matrix(iris[, 1:4])
y <- rep(5, nrow(x))

sfit <- sgdnet(x, y)

coef(sfit)[, 1:10]
#> 5 x 10 sparse Matrix of class "dgCMatrix"
#>    [[ suppressing 10 column names ‘s0’, ‘s1’, ‘s2’ ... ]]
#>                                 
#> (Intercept)  1 1 1 1 1 1 1 1 1 1
#> Sepal.Length . . . . . . . . . .
#> Sepal.Width  . . . . . . . . . .
#> Petal.Length . . . . . . . . . .
#> Petal.Width  . . . . . . . . . .
```

I am currently not allowing constant y `family = "binomial"` because it results in a single class but I think you are right.